openssl-tests/src/raw_key_openssl_interop.rs

```diff
@@ -0,0 +1,499 @@
+    #[derive(Debug)]
+    struct SimpleRpkClientCertVerifier {
+        trusted_spki: Vec<SubjectPublicKeyInfoDer<'static>>,
+        supported_algs: WebPkiSupportedAlgorithms,
+    }
+
+    impl SimpleRpkClientCertVerifier {
+        pub fn new(trusted_spki: Vec<SubjectPublicKeyInfoDer<'static>>) -> Self {
+            let provider = Arc::new(provider::default_provider());
+            Self {
+                trusted_spki,
+                supported_algs: provider
+                    .clone()
+                    .signature_verification_algorithms,
+            }
+        }
+    }
+
+    impl ClientCertVerifier for SimpleRpkClientCertVerifier {
+        fn root_hint_subjects(&self) -> &[DistinguishedName] {
+            &[]
+        }
+
+        fn verify_client_cert(
+            &self,
+            end_entity: &CertificateDer<'_>,
+            _intermediates: &[CertificateDer<'_>],
+            _now: UnixTime,
+        ) -> Result<ClientCertVerified, rustls::Error> {
+            let end_entity_as_spki = SubjectPublicKeyInfoDer::from(end_entity.as_ref());
+            match self
+                .trusted_spki
+                .contains(&end_entity_as_spki)
+            {
+                false => Err(rustls::Error::InvalidCertificate(
+                    CertificateError::UnknownIssuer,
+                )),
+                true => Ok(ClientCertVerified::assertion()),
+            }
+        }
+
+        fn verify_tls12_signature(
+            &self,
+            _message: &[u8],
+            _cert: &CertificateDer<'_>,
+            _dss: &DigitallySignedStruct,
+        ) -> Result<HandshakeSignatureValid, rustls::Error> {
+            Err(rustls::Error::PeerIncompatible(
+                PeerIncompatible::Tls12NotOffered,
+            ))
+        }
+
+        fn verify_tls13_signature(
+            &self,
+            message: &[u8],
+            cert: &CertificateDer<'_>,
+            dss: &DigitallySignedStruct,
+        ) -> Result<HandshakeSignatureValid, rustls::Error> {
+            verify_tls13_signature_with_spki(
+                message,
+                &SubjectPublicKeyInfoDer::from(cert.as_ref()),
+                dss,
+                &self.supported_algs,
+            )
+        }
+
+        fn supported_verify_schemes(&self) -> Vec<SignatureScheme> {
+            self.supported_algs.supported_schemes()
+        }
+
+        fn requires_raw_public_keys(&self) -> bool {
+            true
+        }
+    }
+}
+
+/// Load the raw public key from the file at the given path
+fn load_raw_key(path_to_rpk: &str) -> Vec<SubjectPublicKeyInfoDer<'static>> {
+    let file = File::open(path_to_rpk).expect("cannot open certificate file");
+    let mut reader = BufReader::new(file);
+    rustls_pemfile::public_keys(&mut reader)
+        .map(|result| result.unwrap())
+        .collect()
+}
```

Comment:
This can become: `SubjectPublicKeyInfoDer::from_pem_file(path_to_rpk).expect("canont open public key file")`